frontend/javascripts/oxalis/view/right-border-tabs/segments_tab/segments_view.tsx

```diff
@@ -1520,18 +1499,19 @@ class SegmentsView extends React.Component<Props, State> {
       (segmentId) => `segment-${segmentId}`,
     );
     if (this.props.selectedIds.group != null) {
-      return mappedIdsToKeys.concat(getKeyForGroupId(this.props.selectedIds.group));
+      return mappedIdsToKeys.concat(getNodeKey(GroupTypeEnum.GROUP, this.props.selectedIds.group));
     }
     return mappedIdsToKeys;
   };
 
   getSegmentOrGroupIdsForKeys = (segmentOrGroupKeys: Key[]) => {
     const selectedIds: { segments: number[]; group: number | null } = { segments: [], group: null };
+    const groupPrefix = "Group-";
     segmentOrGroupKeys.forEach((key) => {
       const keyAsString = String(key);
```

Comment:
in which cases is key not a string? in general, I wonder why the `Key` type is used, because it seems like these should always be strings? :thinking_face:

Reply:
It's because the Antd component is using `Key` and I wanted to avoid type conversion. E.g. the tree `onSelect` method needs to take the selected keys as `Key`s. So in the `onSelectTreeItem` method I would need to handle the case that the keys are numbers and convert them to strings. I felt like using `Key` as a type was nice and easy

Reply:
ok, I understand the idea :) the issue is that antd supports numbers and strings for keys, but we never use that in this context. using `Key` everywhere widens the type unnecessarily across various places. for example, if future-me looks into this action in a couple of months, I will definitely wonder why one can pass a number to the expand action (even though, that should never happen and won't really work). so, it's better to catch the "imprecise type" as early as possible and narrow it down.

> So in the onSelectTreeItem method I would need to handle the case that the keys are numbers and convert them to strings.

yes, that would be a good place to narrow the type. I wouldn't even convert from number to string, because this should never happen anyway. if it wasn't an array, I'd throw an exception. however, for an array, that assertion is too costly in my opinion. i'd simply cast from Key to string :) then, you also don't need to do the `String(key)` conversion in line 1511.